library/public/options.h.in

```diff
@@ -114,6 +114,24 @@ public:
    */
   std::pair<std::string, unsigned int> getClosestOption(const std::string& option) const;
 
+  /**
+   *  Returns true if the option is optional else returns false.
+   *  Throws an options::inexistent_exception if option does not exist.
+   */
+  bool isOptional(const std::string& option) const;
+
+  /**
+   *  Resets the option to default value.
+   *  Throws an options::inexistent_exception if option does not exist.
+   */
+  void reset(options& opt, const std::string& name);
+
+  /**
+   *  Unset the option if it is optional.
+   *  Throws an options::inexistent_exception if option does not exist.
+   */
+  void removeValue(options& opt, const std::string& name);
+
```

Comment:
please add tests in library/testing/TestSDKOptions.cxx

Reply:
done